bin/yb-ctl

```diff
@@ -864,8 +946,50 @@ class ClusterControl:
 
     def get_pid(self, daemon_id):
         try:
-            pid = subprocess.check_output(
-                ["pgrep", "-f", self.get_pgrep_regex(daemon_id)]).strip().split()[0]
+            if sys.platform == 'darwin':
+                # -l: Long output.  For pgrep, print the process name in addition to the process ID
+                # for each matching process.  If used in conjunction with -f, print the process ID
+                # and the full argument list for each matching process.  For pkill, display the kill
+                # command used for each process killed.
+                pgrep_full_command_output_arg = '-l'
+            else:
+                # -a, --list-full
+                # List the full command line as well as the process ID.  (pgrep only.)
+                pgrep_full_command_output_arg = '--list-full'
+            pgrep_regex_str = self.get_pgrep_regex(daemon_id)
+            pgrep_output = subprocess.check_output(
+                ["pgrep", pgrep_full_command_output_arg, "-f", pgrep_regex_str]
+            ).strip().decode('utf-8')
+
+            data_dirs_re_match = FS_DATA_DIRS_ARG_RE.search(pgrep_output)
+            if data_dirs_re_match:
+                data_dirs_of_process = data_dirs_re_match.group(1)
+                expected_data_dirs = ','.join(self.get_base_node_dirs(daemon_id.index))
+                if self.args.verbose:
+                    logging.info(
+                            "Data dirs of the running process with daemon id %s: %s",
+                            daemon_id, data_dirs_of_process)
+                if data_dirs_of_process.startswith('"') and data_dirs_of_process.endswith('"'):
```

Comment:
Along similar lines, we should check for single quotes as well.

Reply:
Done.